modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/deployable/AbstractDeployableArtifactDescriptorFactory.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.deployable;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptySet;
+import static java.util.Optional.of;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+
+import org.mule.runtime.api.deployment.meta.MuleDeployableModel;
+import org.mule.runtime.module.artifact.activation.api.ArtifactActivationException;
+import org.mule.runtime.module.artifact.activation.api.descriptor.ArtifactDescriptorFactory;
+import org.mule.runtime.module.artifact.activation.api.deployable.DeployableProjectModel;
+import org.mule.runtime.module.artifact.activation.api.plugin.PluginDescriptorResolver;
+import org.mule.runtime.module.artifact.activation.api.plugin.PluginModelResolver;
+import org.mule.runtime.module.artifact.activation.internal.descriptor.AbstractArtifactDescriptorFactory;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactDescriptorValidatorBuilder;
+import org.mule.runtime.module.artifact.api.descriptor.ArtifactPluginDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDependency;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.ClassLoaderModel;
+import org.mule.runtime.module.artifact.api.descriptor.DeployableArtifactDescriptor;
+import org.mule.tools.api.classloader.model.Artifact;
+import org.mule.tools.api.classloader.model.ArtifactCoordinates;
+
+import com.google.common.collect.ImmutableSet;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Base class to create descriptors of artifacts that are deployable.
+ *
+ * @param <M> type of the artifact model that owns the descriptor.
+ * @param <T> type of descriptor being created.
+ */
+public abstract class AbstractDeployableArtifactDescriptorFactory<M extends MuleDeployableModel, T extends DeployableArtifactDescriptor>
+    extends AbstractArtifactDescriptorFactory<M, T> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractDeployableArtifactDescriptorFactory.class);
+
+  protected final DeployableProjectModel<M> deployableProjectModel;
+  protected final Optional<Properties> deploymentProperties;
```

Comment:
private?

Reply:
Made `deployableProjectModel` private, `deploymentProperties` is used by child classes.